No rule to make target 'tools/objtool/objtool' when compiling kernel module on intel-nuc

For the Intel NUC image, kernel module compilation expects `objtool` to be present inside the `tools/objtool` directory .

But it isn't for `images/intel-nuc/2.26.0%2Brev1.dev/kernel_modules_headers.tar.gz`

```
root@55cd0d3562f5:/usr/src/app# make -C kernel_modules_headers M=/usr/src/app/example_module modules
make: Entering directory '/usr/src/app/kernel_modules_headers'
make[1]: *** No rule to make target 'tools/objtool/objtool', needed by '/usr/src/app/example_module/hello.o'.  Stop.
Makefile:1522: recipe for target '_module_/usr/src/app/example_module' failed
make: *** [_module_/usr/src/app/example_module] Error 2
make: Leaving directory '/usr/src/app/kernel_modules_headers'
root@55cd0d3562f5:/usr/src/app# find kernel_modules_headers | grep objtool
root@55cd0d3562f5:/usr/src/app# 
```

Compiling a recent meta-balena does have the objtool inside the kernel_module_headers tarball..
```
zubairlk@zubair-xps-resin:~/resin/yocto/resin-intel/build/tmp/deploy/images/genericx86-64/temp$ find . | grep objtool
./kernel_modules_headers/tools/objtool
...
```

Possibly linked with minor kernel version bump https://lore.kernel.org/patchwork/patch/762347/

cc @floion @agherzan

--- a/meta-resin-common/recipes-devtools/kernel-modules-headers/kernel-modules-headers.bb
+++ b/meta-resin-common/recipes-devtools/kernel-modules-headers/kernel-modules-headers.bb
@@ -34,6 +34,17 @@ do_configure[noexec] = "1"
 do_compile() {
     mkdir -p kernel_modules_headers
     ${S}/gen_mod_headers ./kernel_modules_headers ${STAGING_KERNEL_DIR} ${DEPLOY_DIR_IMAGE} ${ARCH} ${TARGET_PREFIX} "${CC}" "${HOSTCC}"
+
+    # Sanity test
+    test_arch=$(find kernel_modules_headers/  | xargs file | grep ELF | xargs -I a bash -c 'if ! echo "a" | grep -Fiq "${ARCH}" ; then echo "Did not find ${ARCH}"; fi')
+    if [ ! -z "$test_arch" ]; then
+        bberror "Wrong arch found in ELF files"
+    fi
+    test_interpreter=$(find kernel_modules_headers/  | xargs file | grep ELF | xargs -I a bash -c 'if echo "a" | grep -Fiq "sysroot" ; then echo "Found sysroot in interpreter" ; fi')
+    if [ ! -z "$test_interpreter" ]; then
+        bberror "Sysroot keyword found in interpreter ELF files"
+    fi
+
     tar -czf kernel_modules_headers.tar.gz kernel_modules_headers
     rm -rf kernel_modules_headers
 }

--- a/meta-resin-common/recipes-devtools/kernel-modules-headers/kernel-modules-headers.bb
+++ b/meta-resin-common/recipes-devtools/kernel-modules-headers/kernel-modules-headers.bb
@@ -12,11 +12,15 @@ DEPENDS = " \
     bc-native \
     openssl \
     openssl-native \
+    util-linux-native \
+    elfutils-native \
+    util-linux \
+    elfutils \
     "
 
 SRC_URI = "git://github.com/resin-os/module-headers.git;protocol=https"
 
-SRCREV = "v0.0.11"
+SRCREV = "v0.0.12"
 
 S = "${WORKDIR}/git"
 B = "${WORKDIR}"
@@ -29,7 +33,7 @@ do_configure[noexec] = "1"
 
 do_compile() {
     mkdir -p kernel_modules_headers
-    ${S}/gen_mod_headers ./kernel_modules_headers ${STAGING_KERNEL_DIR} ${DEPLOY_DIR_IMAGE} ${ARCH} ${TARGET_PREFIX} "${HOSTCC}"
+    ${S}/gen_mod_headers ./kernel_modules_headers ${STAGING_KERNEL_DIR} ${DEPLOY_DIR_IMAGE} ${ARCH} ${TARGET_PREFIX} "${CC}" "${HOSTCC}"
     tar -czf kernel_modules_headers.tar.gz kernel_modules_headers
     rm -rf kernel_modules_headers
 }

